main.go

```diff
@@ -126,6 +178,20 @@ func main() {
 			continue
 		}
 
+		// remove data for instances that aren't running
+		for hostname := range timestamps {
+			if strings.HasPrefix(hostname, "ip-") {
+				// convert ES hostnames of the form ip-10-0-0-1 to 10.0.0.1
+				ip := strings.Replace(strings.TrimPrefix("ip-", hostname), "-", ".", -1)
+				running, err := ec2ip.IsRunning(ip)
+				if err != nil {
+					kvlog.ErrorD("ec2-ip-check", kv.M{"error": err.Error()})
+				} else if !running {
+					delete(timestamps, hostname)
```

Comment:
not sure if deleting while iterating over the slice has predictable behavior. 

to be sure, you could allocate another slice and copy valid elements to it

Reply:
https://stackoverflow.com/questions/23229975/is-it-safe-to-remove-selected-keys-from-golang-map-within-a-range-loop

Reply:
👍 til (I googled a different stackoverflow post that was less decisive :P)